packages/web-components/fast-components/src/radio-group/radio-group.ts

```diff
@@ -0,0 +1,174 @@
+
+        if (this.selectedValue === undefined) {
+            radioButtons[0].setAttribute("tabindex", "0");
+        }
+    }
+
+    private getFilteredRadioButtons = (): any[] => {
+        if (this.slottedRadioButtons !== undefined) {
+            return this.slottedRadioButtons.filter((node: Node) => {
+                if (node instanceof HTMLElement) {
+                    return node as HTMLElement;
+                }
+            });
+        } else {
+            return [];
+        }
+    };
+
+    private handleRadioChange = (e: CustomEvent): void => {
+        const changedRadio: FASTRadio = e.target as FASTRadio;
+        if (changedRadio.checked) {
+            this.getFilteredRadioButtons().forEach((radio: FASTRadio) => {
+                if (radio !== changedRadio) {
+                    radio.checked = false;
+                    radio.setAttribute("tabindex", "-1");
+                }
+            });
+            this.selectedRadio = changedRadio;
+            this.selectedValue = changedRadio.value;
+        }
+    };
+
+    private moveToRadioByIndex = (group: any[], index: number) => {
+        const radio: FASTRadio = group[index] as FASTRadio;
+        if (!radio.readOnly) {
+            radio.checked = true;
+        }
+        this.selectedRadio = radio;
+        radio.focus();
+    };
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        const group = this.getFilteredRadioButtons();
+        let index: number = 0;
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                index = group.indexOf(this.selectedRadio) + 1;
+                index = index === group.length ? 0 : index;
+                while (index < group.length) {
+                    if (!group[index].disabled) {
+                        this.moveToRadioByIndex(group, index);
+                        break;
+                    } else if (index === group.indexOf(this.selectedRadio)) {
+                        break;
+                    } else if (index + 1 >= group.length) {
+                        index = 0;
+                    } else {
+                        index += 1;
+                    }
+                }
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowDown:
+                index = group.indexOf(this.selectedRadio) - 1;
+                index = index < 0 ? group.length - 1 : index;
+
+                while (index >= 0) {
+                    if (!group[index].disabled) {
+                        this.moveToRadioByIndex(group, index);
+                        break;
+                    } else if (index === group.indexOf(this.selectedRadio)) {
+                        break;
+                    } else if (index - 1 < 0) {
+                        index = group.length - 1;
+                    } else {
+                        index -= 1;
+                    }
+                }
+                break;
+        }
+    };
+}
```

Comment:
Are we intentionally not supporting the differing keyboard behavior for toolbars as described in the WAI-ARIA spec referenced in the spec? https://w3c.github.io/aria-practices/#for-radio-group-contained-in-a-toolbar

Reply:
Ok I'm assuming, through some research, that I can check to see if the parent has a role of "toolbar", is this how you would expect radio-group to know it's in this toolbar context?

Reply:
I'm not sure without digging deeper into this implementation. It's possible that this would be nested within some divs so it may not be the immediate parent...

Reply:
so this is like looking for direction, it can be all the way up the chain, seems like.

Reply:
I will tidy this up but I think it is now doing the right thing when it's in a toolbar and not in a toolbar. I added a toolbar example in fixtures to test.

Reply:
This could be better but added more comments and cleaned it up